### PR TITLE
aws_byte_cursor_from_array

### PR DESCRIPTION
**Pull request by david-koenig**
*Issue #, if available:*

*Description of changes:*

Add aws_byte_cursor_from_array

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/aws/common/byte_buf.h [hunk 1]
```diff
@@ -51,6 +51,10 @@ static inline struct aws_byte_cursor aws_byte_cursor_from_buf(const struct aws_b
     return (struct aws_byte_cursor){.ptr = buf->buffer, .len = buf->len};
 }
 
+static inline struct aws_byte_cursor aws_byte_cursor_from_array(const uint8_t c_str[], size_t len) {
```
Comment (bdonlan) [inline include/aws/common/byte_buf.h]: Style: I prefer `uint8_t *` in arguments; array syntax comes with some conceptual baggage that doesn't apply in argument lists.

Comment (david-koenig) [reply include/aws/common/byte_buf.h]: I agree 100%. I always write it that way when doing my own code, but I had gone the other way to match the style of aws_byte_buf_from_array in the same file. I will change both of them to match the style you requested.

Incidentally, I prefer to keep spaces on both sides of the * as well when I am declaring a pointer, but it looks like I am losing that battle in this code.

File: include/aws/common/byte_buf.h [hunk 2]
```diff
@@ -51,6 +51,10 @@ static inline struct aws_byte_cursor aws_byte_cursor_from_buf(const struct aws_b
     return (struct aws_byte_cursor){.ptr = buf->buffer, .len = buf->len};
 }
 
+static inline struct aws_byte_cursor aws_byte_cursor_from_array(const uint8_t *c_str, size_t len) {
+    return (struct aws_byte_cursor){.ptr = (uint8_t *)c_str, .len=len};
```
Comment (marcomagdy) [inline include/aws/common/byte_buf.h]: I don't think designated initializers work on msvc (unless I'm missing a flag) https://godbolt.org/g/1a7h2e

(sorry i missed that in earlier reviews)

**Comment by JonathanHenson** [comment]
They do work in vs2015 and later, but I’d recommend against having them in header files.

**Comment by JonathanHenson** [comment]
No, there isn’t but I’ve been using them and building successfully for a long time. In cmake I do specify the language standard though.

**Comment by marcomagdy** [comment]
I just checked it and the compiler flag is `/TC` that's what CMake uses when specifying the language standard.
Anyways, good to know for future godbolting.

**Pull request merged**